runtime/compiler/optimizer/SPMDParallelizer.cpp

```diff
@@ -1427,26 +1451,110 @@ bool TR_SPMDKernelParallelizer::processSPMDKernelLoopForSIMDize(TR::Compilation
    loop->getBlocks(&blocksInLoop);
    ListIterator<TR::Block> blocksIt1(&blocksInLoop);
 
-
    // SIMD_TODO: improve this code
 
    for (TR::Block *nextBlock = blocksIt1.getCurrent(); nextBlock; nextBlock=blocksIt1.getNext())
       {
+      TR_HashTab* entries = new (comp->trStackMemory()) TR_HashTab(comp->trMemory(), stackAlloc);
```

Comment:
I don't think we should create a hash table for each block. I think we need only one that is cleared on every block.

Reply:
Looks like the code assumes there are no extended blocks at this point. So perhaps having a hash table is not really necessary since there will be only one compare and branch and it will always follow an inductions variable increment.  But let's keep the hash table in case we want to handle more general case in the future.